File: day17/solution.js
const _ = require("lodash");
const util = require("../util");

const input = require("fs").readFileSync("./data").toString();

const containers = input.split("\n")
	.map(container => parseInt(container, 10)).sort((a, b) => a - b);

let sumTo150 = util.combinations(containers, 4)
	.filter(containers => util.sum(containers) === 150);

console.log(sumTo150.length);

File: day17/solution.py
import itertools

data = open("data", "r").read()

containers = map(int, data.split("\n"))

part1 = []
minLength = None
for length in range(len(containers)):
	combinations = itertools.combinations(containers, length)
	combinations = filter(lambda containers: sum(containers) == 150, combinations)

	part1 += combinations

	if minLength is None and len(combinations) > 0:
		minLength = length


print "Combinations (Part 1):", len(part1)

part2 = filter(lambda containers: len(containers) == minLength, part1)

print "Minimum Length Combinations (Part 2):", len(part2)
